R/gateUtil.R

```diff
@@ -118,3 +118,44 @@ commonGateCreate = function(body, name, gid,
     return(list(gate = gateResp))
   }
 }
+
+# Assigns common properties to body of compound gate , then makes the request.
+compoundGateCreate = function(body, names, gid, gids,
+                            experimentId,
+                            parentPopulationId, parentPopulation,
+                            tailoredPerFile, fcsFileId, fcsFile,
+                            createPopulation) {
+
+  checkDefined(experimentId)
+
+  parentPopulationId = parsePopulationArgs(parentPopulationId, parentPopulation,
+    experimentId)
+
+  body = c(body, list(
+    parentPopulationId = jsonlite::unbox(parentPopulationId),
+    names = names,
+    gid = jsonlite::unbox(gid)
+  ))
+
+  body = parseFcsFileArgs(body, tailoredPerFile, fcsFileId, fcsFile, experimentId)
+
+  body = jsonlite::toJSON(body, null = "null", digits = NA)
+  path = paste("experiments", experimentId, "gates", sep = "/")
+  gateResp = basePost(path, body, list())
+  
+  if (createPopulation) {
```

Comment:
Another later TODO: Our API can do this automatically by passing `createPopulation=true` to the `POST gates` endpoint.

Reply:
To clarify, do you mean that the `autoCreatePopulation` function inside this `if (createPouplation)` statement can be skipped? I.e. is there server-side logic that can respond to the `createPopulation` arg, instead of doing it in R?

Reply:
Correct. I'd also make that change in a separate PR for all gate fns at once though.